File: TheOpenMathLibrary.Engineering/FluidDynamics.cs
namespace TheOpenMathLibrary.Engineering
{
    public class FluidDynamics
    {
        //flow velocity function
        public static double FlowVelocity(double flowRate, double pipeDiameter)
        {
            double velocity = 0;
            velocity = (4 * flowRate) / (3.14159 * pipeDiameter * pipeDiameter);
            return velocity;
        }

        //velocity pseudovector function
        public static double VelocityPseudovector(double velocity, double radius)
        {
            double pseudovector = 0;
            pseudovector = velocity * radius;
            return pseudovector;
        }

        //volume flux function
        public static double VolumeFlux(double flowRate, double radius)
        {
            double volumeFlux = 0;
            volumeFlux = flowRate / (3.14159 * radius * radius);
            return volumeFlux;
        }

        //mass current per volume function
        public static double MassCurrentPerVolume(double massCurrent, double volume)
        {
            double massCurrentPerVolume = 0;
            massCurrentPerVolume = massCurrent / volume;
            return massCurrentPerVolume;
        }

        //mass flow rate function
        public static double MassFlowRate(double density, double volumeFlux)
        {
            double massFlowRate = 0;
            massFlowRate = density * volumeFlux;
            return massFlowRate;
        }

        //mass current density function
        public static double MassCurrentDensity(double density, double velocity)
        {
            double massCurrentDensity = 0;
            massCurrentDensity = density * velocity;
            return massCurrentDensity;
        }

        //momentum current desnity function
        public static double MomentumCurrentDensity(double density, double velocity)
        {
            double momentumCurrentDensity = 0;
            momentumCurrentDensity = density * velocity * velocity;
            return momentumCurrentDensity;
        }

        //pressure gradient function
        public static double PressureGradient(double pressure, double distance)
        {
            double pressureGradient = 0;
            pressureGradient = pressure / distance;
            return pressureGradient;
        }

        //Buoyancy equations
        public static double BuoyancyForce(double density, double volume, double gravity)
        {
            double buoyancyForce = 0;
            buoyancyForce = density * volume * gravity;
            return buoyancyForce;
        }

        //Bernoulli's equation
        public static double BernoullisEquation(double pressure, double density, double velocity, double height, double gravity)
        {
            double bernoullisEquation = 0;
            bernoullisEquation = pressure + (density * velocity * velocity) + (density * gravity * height);
            return bernoullisEquation;
        }

        //Eulers Equations
        public static double EulersEquations(double pressure, double density, double velocity, double height, double gravity)
        {
            double eulersEquations = 0;
            eulersEquations = pressure + (density * velocity * velocity) + (density * gravity * height);
            return eulersEquations;
        }

        //Convective acceleration function
        public static double ConvectiveAcceleration(double velocity, double acceleration)
        {
            double convectiveAcceleration = 0;
            convectiveAcceleration = velocity * acceleration;
            return convectiveAcceleration;
        }

        //Navier stokes equations
        public static double NavierStokesEquations(double pressure, double density, double velocity, double height, double gravity)
        {
            double navierStokesEquations = 0;
            navierStokesEquations = pressure + (density * velocity * velocity) + (density * gravity * height);
            return navierStokesEquations;
        }
    }
}

File: Go/TheOpenMathLibrary/Engineering/FluidDynamics.go
package engineering

// FluidDynamics contains methods for calculating various properties of fluid dynamics.
type FluidDynamics struct{}

// FlowVelocity calculates the flow velocity given a flow rate and pipe diameter.
func FlowVelocity(flowRate, pipeDiameter float64) float64 {
	return (4 * flowRate) / (3.14159 * pipeDiameter * pipeDiameter)
}

// VelocityPseudovector calculates the velocity pseudovector given a velocity and radius.
func VelocityPseudovector(velocity, radius float64) float64 {
	return velocity * radius
}

// VolumeFlux calculates the volume flux given a flow rate and radius.
func VolumeFlux(flowRate, radius float64) float64 {
	return flowRate / (3.14159 * radius * radius)
}

// MassCurrentPerVolume calculates the mass current per volume given a mass current and volume.
func MassCurrentPerVolume(massCurrent, volume float64) float64 {
	return massCurrent / volume
}

// MassFlowRate calculates the mass flow rate given a density and volume flux.
func MassFlowRate(density, volumeFlux float64) float64 {
	return density * volumeFlux
}

// MassCurrentDensity calculates the mass current density given a density and velocity.
func MassCurrentDensity(density, velocity float64) float64 {
	return density * velocity
}

// MomentumCurrentDensity calculates the momentum current density given a density and velocity.
func MomentumCurrentDensity(density, velocity float64) float64 {
	return density * velocity * velocity
}

// PressureGradient calculates the pressure gradient given a pressure and distance.
func PressureGradient(pressure, distance float64) float64 {
	return pressure / distance
}

// BuoyancyForce calculates the buoyancy force given a density, volume, and gravity.
func BuoyancyForce(density, volume, gravity float64) float64 {
	return density * volume * gravity
}

// BernoullisEquation calculates Bernoulli's equation given a pressure, density, velocity, height, and gravity.
func BernoullisEquation(pressure, density, velocity, height, gravity float64) float64 {
	return pressure + (density * velocity * velocity) + (density * gravity * height)
}

// EulersEquations calculates Euler's equations given a pressure, density, velocity, height, and gravity.
func EulersEquations(pressure, density, velocity, height, gravity float64) float64 {
	return pressure + (density * velocity * velocity) + (density * gravity * height)
}

// ConvectiveAcceleration calculates the convective acceleration given a velocity and acceleration.
func ConvectiveAcceleration(velocity, acceleration float64) float64 {
	return velocity * acceleration
}

// NavierStokesEquations calculates the Navier-Stokes equations given a pressure, density, velocity, height, and gravity.
func NavierStokesEquations(pressure, density, velocity, height, gravity float64) float64 {
	return pressure + (density * velocity * velocity) + (density * gravity * height)
}
